### DISCUSSION TITLE
FIP: Subcommunities

### DISCUSSION THREAD

**Discussion by davidfurlong** [farcasterxyz/protocol#92] [FIP Stage 1: Ideas]
**Problem**

Farcaster has many subcommunities and conversations around hobbies, topics, sports or DAOs, but there's no easy way for people to find conversations around their communities nor post directly to a specific community.

An example is Purple (as in the DAO) that wants a way to have conversations around purple on Farcaster. Clients don't have a good way of surfacing or making Purple specific conversations possible in an interoperable way on Farcaster, as it isn't possible to reliably classify most casts that are about Purple as about Purple, particularly if they don't mention Purple directly, and even if they do it's not trivial to detect whether it's a reference to the color or the DAO.

There are many Farcaster only communities that don't use Farcaster to communicate, and while there is a place for external community apps, I think it's a missed opportunity. For example Purple (telegram + discord), Farcaster dev chat (telegram), Farcaster life (telegram), Farcaster LA (telegram), Farcaster video clients (telegram). 

Current ways to do achieve subcommunities without this proposal:
- Add an emoji to the end of every cast, corresponding to the topic. Doesn't scale beyond the set of emojis and isn't ideal
- Make everything a reply to a single thread
- Have clients try to detect subcommunities just from the cast context
- Have users use hashtags or add other strings to casts
- Use another tool, like discord or telegram (most common)

**Solution**

For context, https://github.com/farcasterxyz/protocol/discussions/71 unlocks casts that embed or reply directly to URIs.

I'm suggesting a URI scheme for tagging casts that makes subcommunities on Farcaster possible. Clients can make it easy for people to see all the casts that have a `parentUri` or `embed` of for example `tag://formula1`, and provide a way to for people to create a cast in that community. With Purple this might already be possible by agreeing to reply to the `CAIP` of the NFT collection of the DAO. 

I am not entirely clear on the exact implementation that is best, but for a top level, short, flexible URI namespace `tag://` may be an option. A better option may be to just use NFTs as subcommunities, and anyone who wants to start one, like "Formula 1" needs to mint an NFT and try to get enough people to participate in this subcommunity; that probably wouldn't require any protocol changes or this FIP.

Clients could surface the most popular (by number of recent casts) subcommunities and perhaps have a power-user interface for people to try to start new communities and actually configure the initial `parentUri` url. A Purple token holder subcommunity could be created by clients by showing a feed of casts replying to that subcommunity and only include casters that hold the Purple token. 

Additional 2nd order effects: it makes it easy for clients to detect what people are interested in, as it's possible to tell which subcommunities people are participating in.

**Open questions:**
1. How do you disambiguate between two words that have two different communities - for example "Cars" _the movie_ and _the type of vehicle_? An example solution is from wikipedia which adds the context (film). It may end up being something like `tag://films/cars` and `tag://cars`. Reddit allows anyone to create a subcommunity and register a handle for it by being first but this seems unlikely to be a good solution. A promising way to do this currently would be to say anyone can create a farcaster subcommunity by minting an NFT, and then to have a subcommunity you reply to that NFT via https://github.com/farcasterxyz/protocol/discussions/71
2. Is there an open standard format that can be used for this? perhaps [schema.org](https://schema.org/docs/schemas.html) types serialized as strings? 
3. The categories probably shouldn't be case sensitive, but there may be additional metadata helpful for clients to display a human readable name and image. This is where the NFT method would be helpful.
4. Is this the best way for clients to indicate "Narrowcasting" or "Nearcasting" a cast to be distributed to only specific people or subcommunities for whom this cast is relevant, so as to not spam your followers and default feeds?
5. What's the best way for users to indicate preference for seeing these casts in their feeds? Is there a way to subscribe or follow subcommunities? If an NFT was used for the subcommunity as above, then minting the NFT might be a natural way of doing this. However this is possibly harmful, as users could setup subcommunities just in order to rent-seek through charging for the NFT.
6. By using `embeds` for the target, a cast becomes ambigious to whether someone wants to share more broadly about a subcommunity or more narrowly. By using the `parentUri`, it only allows for one target subcommunity or tag. What if someone wants to post something relevant to more than one subcommunity, perhaps suggesting a collab between both, to both? Is there a better abstraction to allow for more flexibility in this?  

EDIT:
https://github.com/farcasterxyz/protocol/discussions/85 Allows for subscribing or following these subcommunities

**Comment by codewithmilo** [comment]
I've been thinking about tags a ton as well, and I think they would be add a ton of value to the protocol. Not just by way of detailing cast content, which in some ways is a sort of hashtag-convenience-replacement, but also metadata about the sending client or context. Clients can much more easily sort through the graph for content by tags, but also could do things like prioritize content generated from themselves or even only show casts from the client.

Thinking of implementation, I can imagine a field similar to `reactions` that has a very limited set of string tags, e.g. up to 5 of max 10 character strings (or a single string with a known delimiter, whatever would take up the least space).

**Reply by sigmoidfractal** [reply]
I like the idea of reactions, and imho it should only need 2 to match reddit's functionality (subcommunity + type of post). the question I would have is how should the subcommunity be forked, and how could the "reddit repost" be implemented? there's already bot accounts like surveycaster, wondering if that also replaces a potential string tag.

**Reply by codewithmilo** [reply]
Hmm, would you mind elaborating on the reddit repost? I'm not familiar with that.

**Reply by sigmoidfractal** [reply]
(as opposed to the quote tweet) I think that any given post on reddit might overlap with subreddits of a relevant topic, in addition to other communities effectively competing over the same topic. so as I see it, the "reddit repost" is the simplest no-commentary method of adding links in the graph. then again, this could be abused if any subcommunity is sufficiently splintered.
the more I think about it, the more bots could be leaned on for passive subcommunity moderation via filtered recasting.
e.g. Alice credibly casts a publication about the James Webb telescope imaging a blackhole, HawkingBot recasts to a "physics" subcommunity, GalileoBot to a "telescopy" subcommunity. while Bob might spam corporate political content to "news", the TarbellBot might filter it out on the basis of content or conduct. 
Also, kind of make me wonder what other tags are needed for these sorts of automated "recast-to-subcommunity", like a hot take modifier.

**Comment by psatyajeet** [comment]
Love the concept of tags. Would make it easier to create feed-based apps like sportscaster (kmac idea) and to make Discove more reliable in tying together related casts. Am wholly supportive of this direction. 

Farcaster needs to attract niche communities and enable lively conversation within them to really grow

**Comment by whatrocks** [comment]
> * Add an emoji to the end of every cast, corresponding to the topic. Doesn't scale beyond the set of emojis and isn't ideal

This is akin to what I've done with my fling "pattern": https://gist.github.com/whatrocks/b4b7c306b307c3c707f0d6256fe9afb0 which is about using Farcaster to create a "social" daily log service. I've been using it and a few others have tried, but the seeming top concern is that folks don't want to "clutter" their own feed (and thus the feeds of their followers) with these non-Twitter-like casts.

**Comment by varunsrin** [comment]
With FIP-2 channels, I think we achieve *most* of what this proposal set out to do. @davidfurlong lmk if you want to keep this active or think we can close it in favor of FIP-2.

**Reply by davidfurlong** [reply]
yeah I'm happy to close this
